fmn/backends/datagrepper.py

```diff
@@ -0,0 +1,38 @@
+import logging
+from typing import Any, AsyncIterator
+
+from httpx import AsyncClient
+
+from ..core.util import make_synchronous
+from .base import APIClient
+
+log = logging.getLogger(__name__)
+
+
+class DatagrepperAsyncProxy(APIClient):
+    """Proxy for the Datagrepper API endpoints used in FMN"""
+
+    API_VERSION = "v2"
+
+    payload_field = "raw_messages"
+
+    def __init__(self, base_url: str):
+        self.client = AsyncClient(base_url=f"{base_url.rstrip('/')}/{self.API_VERSION}", timeout=0)
```

Comment:
Actually the timeout should be `None` to disable it. When it's zero it timeouts instantly :-)

Reply:
Ahh, good catch.